# Leega - Desafio Análise de Dados
 ## Base de dados - Base Space_corrected.CSV

Este desafio Responde as principais questões
*   Qual é o total de missões registradas?
*   Quantas foram bem sucedidas?
*   Custo das missões por ano?
*   Que dado interessante você poderia acrescentar ao estudo?






In [172]:
import pandas as pd
df = pd.read_csv('Space.csv')

In [173]:
df.columns

Index(['Unnamed: 0.1', 'Unnamed: 0', 'Company Name', 'Location', 'Datum',
       'Detail', 'Status Rocket', ' Rocket', 'Status Mission'],
      dtype='object')

In [174]:
df.head(15)

,Unnamed: 0.1,Unnamed: 0,Company Name,Location,Datum,Detail,Status Rocket,Rocket,Status Mission
0,0,0,SpaceX,"LC-39A, Kennedy Space Center, Florida, USA","Fri Aug 07, 2020 05:12 UTC",Falcon 9 Block 5 | Starlink V1 L9 & BlackSky,StatusActive,50.0,Success
1,1,1,CASC,"Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...","Thu Aug 06, 2020 04:01 UTC",Long March 2D | Gaofen-9 04 & Q-SAT,StatusActive,29.75,Success
2,2,2,SpaceX,"Pad A, Boca Chica, Texas, USA","Tue Aug 04, 2020 23:57 UTC",Starship Prototype | 150 Meter Hop,StatusActive,NaN,Success
3,3,3,Roscosmos,"Site 200/39, Baikonur Cosmodrome, Kazakhstan","Thu Jul 30, 2020 21:25 UTC",Proton-M/Briz-M | Ekspress-80 & Ekspress-103,StatusActive,65.0,Success
4,4,4,ULA,"SLC-41, Cape Canaveral AFS, Florida, USA","Thu Jul 30, 2020 11:50 UTC",Atlas V 541 | Perseverance,StatusActive,145.0,Success
5,5,5,CASC,"LC-9, Taiyuan Satellite Launch Center, China","Sat Jul 25, 2020 03:13 UTC","Long March 4B | Ziyuan-3 03, Apocalypse-10 & N...",StatusActive,64.68,Success
6,6,6,Roscosmos,"Site 31/6, Baikonur Cosmodrome, Kazakhstan","Thu Jul 23, 2020 14:26 UTC",Soyuz 2.1a | Progress MS-15,StatusActive,48.5,Success
7,7,7,CASC,"LC-101, Wenchang Satellite Launch Center, China","Thu Jul 23, 2020 04:41 UTC",Long March 5 | Tianwen-1,StatusActive,NaN,Success
8,8,8,SpaceX,"SLC-40, Cape Canaveral AFS, Florida, USA","Mon Jul 20, 2020 21:30 UTC",Falcon 9 Block 5 | ANASIS-II,StatusActive,50.0,Success
9,9,9,JAXA,"LA-Y1, Tanegashima Space Center, Japan","Sun Jul 19, 2020 21:58 UTC",H-IIA 202 | Hope Mars Mission,StatusActive,90.0,Success


# Correções
- Index Duplicado

In [175]:
# Visualizando o Dataframe e dropando colunas de index.
df = df.loc[:, ~df.columns.str.contains('^Unnamed')]
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Company Name    4324 non-null   object
 1   Location        4324 non-null   object
 2   Datum           4324 non-null   object
 3   Detail          4324 non-null   object
 4   Status Rocket   4324 non-null   object
 5    Rocket         964 non-null    object
 6   Status Mission  4324 non-null   object
dtypes: object(7)
memory usage: 236.6+ KB


# Tradução Mapeamento Inglês -> Português

In [176]:
# Dicionário de tradução
traducao_colunas = {
    'Company Name': 'Nome_da_Empresa',
    'Location': 'Localizacao',
    'Datum': 'Data_Lancamento',
    'Detail': 'Detalhes_da_Missao',
    'Status Rocket': 'Status_do_Foguete',
    ' Rocket': 'Custo_da_Missao',
    'Status Mission': 'Status_da_Missao'
}

# Renomeando as colunas
df = df.rename(columns=traducao_colunas)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Nome_da_Empresa     4324 non-null   object
 1   Localizacao         4324 non-null   object
 2   Data_Lancamento     4324 non-null   object
 3   Detalhes_da_Missao  4324 non-null   object
 4   Status_do_Foguete   4324 non-null   object
 5   Custo_da_Missao     964 non-null    object
 6   Status_da_Missao    4324 non-null   object
dtypes: object(7)
memory usage: 236.6+ KB


# Validação de Textos/Categorias

## Correções
- Armée de l'Air está escrito errado

In [177]:
# Corrigindo o caractere corrompido no nome da agência francesa
df['Nome_da_Empresa'] = df['Nome_da_Empresa'].replace("Arm??e de l'Air", "Armée de l'Air")


In [178]:
# Verificando os nomes das Empresas
print("\n LISTA DE EMPRESAS ÚNICAS:")
print(sorted(df['Nome_da_Empresa'].unique()))
print("-" * 60)

# Verificando o Status do Foguete
print("\n STATUS DO FOGUETE:")
print(df['Status_do_Foguete'].unique())
print("-" * 60)

# Verificando o Status da Missão
print("\n STATUS DA MISSAO:")
print(df['Status_da_Missao'].unique())
print("-" * 60)

#  Verificando os Modelos de Foguete
print("\n TOP 15 MODELOS DE FOGUETES MAIS FREQUENTES:")
print(df['Detalhes_da_Missao'].value_counts().head(15))
print("="*60)


 LISTA DE EMPRESAS ÚNICAS:
['AEB', 'AMBA', 'ASI', 'Arianespace', "Armée de l'Air", 'Blue Origin', 'Boeing', 'CASC', 'CASIC', 'CECLES', 'CNES', 'Douglas', 'EER', 'ESA', 'Eurockot', 'ExPace', 'Exos', 'General Dynamics', 'IAI', 'ILS', 'IRGC', 'ISA', 'ISAS', 'ISRO', 'JAXA', 'KARI', 'KCST', 'Khrunichev', 'Kosmotras', 'Land Launch', 'Landspace', 'Lockheed', 'MHI', 'MITT', 'Martin Marietta', 'NASA', 'Northrop', 'OKB-586', 'OneSpace', 'RAE', 'RVSN USSR', 'Rocket Lab', 'Roscosmos', 'SRC', 'Sandia', 'Sea Launch', 'SpaceX', 'Starsem', 'ULA', 'US Air Force', 'US Navy', 'UT', 'VKS RF', 'Virgin Orbit', 'Yuzhmash', 'i-Space']
------------------------------------------------------------

 STATUS DO FOGUETE:
['StatusActive' 'StatusRetired']
------------------------------------------------------------

 STATUS DA MISSAO:
['Success' 'Failure' 'Prelaunch Failure' 'Partial Failure']
------------------------------------------------------------

 TOP 15 MODELOS DE FOGUETES MAIS FREQUENTES:
Detalhes_da_Missa

## Feature Engineering
Separação e extração de variáveis textuais e temporais  (Localizacao, Data_Lancamento e Detalhes_da_Missao) em novas colunas atômicas, estruturadas e tipadas (Pais, Ano, Modelo_Foguete, etc.) para otimização da análise, seguida do descarte de dados redundantes.


## Decomposição dos Detalhes da Missão (Modelo_Foguete e Carga_Util)
O que são:  
- Modelo_Foguete: É o veículo lançador, a máquina física responsável por vencer a gravidade da Terra (ex: Falcon 9, Ariane 5, Soyuz).

- Carga_Util (Payload): É o objeto ou tripulação que está sendo transportado para o espaço (ex: um satélite de internet da Starlink, o robô Perseverance enviado para Marte, ou uma cápsula com astronautas).

Por que separamos:
Na base bruta, essas duas informações vinham coladas na mesma linha de texto, separadas por uma barra vertical (|). Se mantivéssemos os dados juntos, seria impossível responder a perguntas cruciais de negócio de forma isolada, como: "Qual é o modelo de foguete mais utilizado na história?" ou "Quantos satélites de pesquisa foram enviados?". A separação transforma um bloco de texto confuso em duas métricas limpas, permitindo filtros exatos no Dashboard e análises estatísticas individuais para cada componente da missão.

In [179]:
# Tratamento da Coluna de Localização
df['Pais'] = df['Localizacao'].str.split(',').str[-1].str.strip()
df['Estado_Regiao'] = df['Localizacao'].str.split(',').str[-2].str.strip()

# Tratamento da Coluna de Data e Hora
df['Data_Auxiliar'] = pd.to_datetime(df['Data_Lancamento'].str.replace('UTC', ''), errors='coerce')

df['Ano'] = df['Data_Auxiliar'].dt.year
df['Mes'] = df['Data_Auxiliar'].dt.month
df['Dia'] = df['Data_Auxiliar'].dt.day
df['Hora'] = df['Data_Auxiliar'].dt.hour

# Garantindo o preenchimento do ano caso a conversão de data falhe em linhas antigas
df['Ano'] = df['Ano'].fillna(df['Data_Lancamento'].str.extract(r'([1-2][0-9]{3})')[0].astype(float))

# Corrigindo a formatação para números inteiros puros
colunas_tempo = ['Ano', 'Mes', 'Dia', 'Hora']
for col in colunas_tempo:
    df[col] = df[col].astype('Int64')

# Removendo a coluna auxiliar
df = df.drop(columns=['Data_Auxiliar'])

# Tratamento dos Detalhes da Missão
df[['Modelo_Foguete', 'Carga_Util']] = df['Detalhes_da_Missao'].str.split('|', n=1, expand=True)

df['Modelo_Foguete'] = df['Modelo_Foguete'].str.strip()
df['Carga_Util'] = df['Carga_Util'].str.strip()

# Apagando a coluna original já que as informações foram extraídas
df = df.drop(columns=['Detalhes_da_Missao','Data_Lancamento', 'Localizacao'])


# Padronização de países para gráficos de Mapas

In [180]:
# Dicionário de padronização de países para os mapas dos Dashboards
# Corrige abreviações e locais marítimos que dariam erro no mapa
mapa_paises = {
    'USA': 'United States',
    'Russia': 'Russia',
    'Kazakhstan': 'Kazakhstan',
    'China': 'China',
    'France': 'France',
    'Japan': 'Japan',
    'India': 'India',
    'Iran': 'Iran',
    'Israel': 'Israel',
    'New Zealand': 'New Zealand',
    'South Korea': 'South Korea',
    'North Korea': 'North Korea',
    'Brazil': 'Brazil',
    'Australia': 'Australia',
    'Kenya': 'Kenya',

    # Padronização de Lançamentos Marítimos
    'Barents Sea': 'International Waters',
    'Pacific Ocean': 'International Waters',
    'Yellow Sea': 'International Waters',

    # Correção de erros da base original
    'Shahrud Missile Test Site': 'Iran',
    'New Mexico': 'United States',
    'Pacific Missile Range Facility': 'United States', # Fica no Havaí
    'Gran Canaria': 'Spain'                            # Pertence à Espanha
}

# Aplicando a padronização
df['Pais_Mapa'] = df['Pais'].replace(mapa_paises)

# Comparação de valores/países

In [181]:
# Lista todos os valores únicos que existem na coluna original
print("Países na base original:")
print(df['Pais'].unique())

# Lista todos os valores únicos após a nossa padronização
print("Países prontos para o Mapa:")
print(df['Pais_Mapa'].unique())

Países na base original:
['USA' 'China' 'Kazakhstan' 'Japan' 'Israel' 'New Zealand' 'Russia'
 'Shahrud Missile Test Site' 'France' 'Iran' 'India' 'New Mexico'
 'Yellow Sea' 'North Korea' 'Pacific Missile Range Facility'
 'Pacific Ocean' 'South Korea' 'Barents Sea' 'Brazil' 'Gran Canaria'
 'Kenya' 'Australia']
Países prontos para o Mapa:
['United States' 'China' 'Kazakhstan' 'Japan' 'Israel' 'New Zealand'
 'Russia' 'Iran' 'France' 'India' 'International Waters' 'North Korea'
 'South Korea' 'Brazil' 'Spain' 'Kenya' 'Australia']


# VERIFICAÇÃO DE VALORES NULOS

In [182]:

valores_nulos = df.isna().sum()
# Mostrando apenas as colunas que têm algum valor vazio para dar destaque
colunas_com_nulos = valores_nulos[valores_nulos > 0]

if not colunas_com_nulos.empty:
    print(colunas_com_nulos)
else:
    print("Nenhum valor nulo encontrado na base!")

print(f"Total de linhas na base: {len(df)}")

Custo_da_Missao    3360
Mes                 126
Dia                 126
Hora                126
dtype: int64
Total de linhas na base: 4324


## Estratégia de Tratamento de Dados Faltantes (Missing Values)
Ao verificar os valores nulos da nossa base, identificamos dois cenários distintos que exigem tratamentos diferentes:

1. Dados Temporais (Mês, Dia e Hora)
Temos 126 registros vazios. Tratam-se de missões históricas antigas (décadas de 50, 60 e 70) onde essas informações não foram documentadas com precisão na época.

Decisão: Manter esses valores em branco (Nulos). Não faremos preenchimentos para preservar a veracidade histórica dos dados.

2. Dados Financeiros (Custo da Missão)
Identificamos 3.360 valores vazios (cerca de 78% da base). Remover essas linhas destruiria nosso dataset. Portanto, traçamos uma estratégia estatística para preencher essas lacunas sem distorcer a realidade do mercado.

 **O Erro da Primeira Tentativa e a Mudança de Planejamento
Inicialmente, tentamos aplicar uma abordagem padrão combinando Imputação Condicional pela Empresa (Plano A) com um Fallback pela Mediana Global (Plano B). No entanto, a validação dos dados revelou duas grandes distorções (outliers):**

O caso RVSN USSR (União Soviética): Como a agência soviética possuía um foguete histórico (Energia) com custo registrado absurdamente alto (5.000 milhões), a mediana da empresa ficou travada nesse teto. A imputação simples fez com que todos os seus lançamentos comuns e antigos herdassem erroneamente esse custo bilionário, inflando drasticamente a média histórica.

O "Efeito 450.0": A mediana global pura da base estava fixada em 450 milhões devido ao peso de grandes agências modernas. Isso fez com que empresas menores que não tinham nenhum dado financeiro (como a AEB - Agência Espacial Brasileira) recebessem valores irreais por seus lançamentos.

## Novo Planejamento Adotado (Imputação Refinada por Camadas)
Para corrigir essas distorções e blindar nossa análise, mudamos o planejamento para uma estratégia hierárquica muito mais precisa e contextualizada:

Camada de Segurança (Tratamento de Outliers): Identificamos e isolamos manualmente o impacto da RVSN USSR, aplicando um teto de custo realista (150 milhões) para suas missões comuns que estavam sem registro.

- Imputação Condicional por Empresa (Plano A): Mantivemos o preenchimento dos custos vazios usando a mediana da própria empresa responsável, respeitando o perfil operacional de gastos de cada agência (como a diferença entre os altos custos da NASA e os custos comerciais da SpaceX).

- Imputação Condicional por Ano/Década (Plano B - Novo Fallback): Em vez de usar uma mediana global genérica para empresas sem dados, o código agora busca a mediana das missões ocorridas naquele mesmo ano. Isso respeita a evolução da economia e da tecnologia (foguetes antigos eram mais baratos que foguetes modernos).

- Mediana Global Ajustada (Plano C): Como último recurso para casos raros e isolados, aplicamos a mediana global, porém filtrando e ignorando os grandes outliers numéricos da base para manter a integridade dos dados.

Dessa forma, os dados pós-imputação passaram com sucesso na validação, apresentando médias perfeitamente coerentes com a realidade histórica e econômica.

In [183]:
# Certificando que a coluna é numérica
df['Custo_da_Missao'] = df['Custo_da_Missao'].astype(str).str.replace(',', '', regex=False).str.strip()
df['Custo_da_Missao'] = pd.to_numeric(df['Custo_da_Missao'], errors='coerce')

# [Ajuste de Segurança] A RVSN USSR tem um outlier de 5000 que distorce a história.
# Vamos limitar o custo máximo dela à mediana das outras agências da mesma época se for nulo.
df.loc[(df['Nome_da_Empresa'] == 'RVSN USSR') & (df['Custo_da_Missao'].isna()), 'Custo_da_Missao'] = 150.0 # Custo estimado realista de um foguete soviético padrão

# Imputação por Empresa (Plano A)
mediana_por_empresa = df.groupby('Nome_da_Empresa')['Custo_da_Missao'].transform('median')
df['Custo_da_Missao'] = df['Custo_da_Missao'].fillna(mediana_por_empresa)

# Imputação por Ano/Década (Plano B - Muito mais preciso que a global pura)
mediana_por_ano = df.groupby('Ano')['Custo_da_Missao'].transform('median')
df['Custo_da_Missao'] = df['Custo_da_Missao'].fillna(mediana_por_ano)

# Se ainda sobrar algum nulo (empresas isoladas em anos sem dados), usamos a mediana global baixa
mediana_geral_ajustada = df[df['Custo_da_Missao'] < 1000]['Custo_da_Missao'].median() # Ignora o outlier de 5000 no cálculo geral
df['Custo_da_Missao'] = df['Custo_da_Missao'].fillna(mediana_geral_ajustada)

print(df.groupby('Nome_da_Empresa')['Custo_da_Missao'].mean().sort_values(ascending=False).head(5))

Nome_da_Empresa
NASA              495.467980
Arianespace       189.767025
Boeing            164.683824
RVSN USSR         155.458638
Armée de l'Air    150.000000
Name: Custo_da_Missao, dtype: float64


In [184]:
valores_nulos = df.isna().sum()
# Mostrando apenas as colunas que têm algum valor vazio para dar destaque
colunas_com_nulos = valores_nulos[valores_nulos > 0]

if not colunas_com_nulos.empty:
    print(colunas_com_nulos)
else:
    print("Nenhum valor nulo encontrado na base!")

print(f"Total de linhas na base: {len(df)}")

Mes     126
Dia     126
Hora    126
dtype: int64
Total de linhas na base: 4324


In [185]:
print(df.info())
display(df.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Nome_da_Empresa    4324 non-null   object 
 1   Status_do_Foguete  4324 non-null   object 
 2   Custo_da_Missao    4324 non-null   float64
 3   Status_da_Missao   4324 non-null   object 
 4   Pais               4324 non-null   object 
 5   Estado_Regiao      4324 non-null   object 
 6   Ano                4324 non-null   Int64  
 7   Mes                4198 non-null   Int64  
 8   Dia                4198 non-null   Int64  
 9   Hora               4198 non-null   Int64  
 10  Modelo_Foguete     4324 non-null   object 
 11  Carga_Util         4324 non-null   object 
 12  Pais_Mapa          4324 non-null   object 
dtypes: Int64(4), float64(1), object(8)
memory usage: 456.2+ KB
None


,Nome_da_Empresa,Status_do_Foguete,Custo_da_Missao,Status_da_Missao,Pais,Estado_Regiao,Ano,Mes,Dia,Hora,Modelo_Foguete,Carga_Util,Pais_Mapa
0,SpaceX,StatusActive,50.00,Success,USA,Florida,2020,8,7,5,Falcon 9 Block 5,Starlink V1 L9 & BlackSky,United States
1,CASC,StatusActive,29.75,Success,China,Jiuquan Satellite Launch Center,2020,8,6,4,Long March 2D,Gaofen-9 04 & Q-SAT,China
2,SpaceX,StatusActive,56.50,Success,USA,Texas,2020,8,4,23,Starship Prototype,150 Meter Hop,United States
3,Roscosmos,StatusActive,65.00,Success,Kazakhstan,Baikonur Cosmodrome,2020,7,30,21,Proton-M/Briz-M,Ekspress-80 & Ekspress-103,Kazakhstan
4,ULA,StatusActive,145.00,Success,USA,Florida,2020,7,30,11,Atlas V 541,Perseverance,United States


# CODIFICAÇÃO NUMÉRICA (ENCODING)

In [186]:
# Transformando as Categorias em Números (Mapeamento) ---

# Mapeamento do Status da Missão (Escala Ordinal de Sucesso)
mapa_status_missao = {
    'Failure': 0,
    'Prelaunch Failure': 1,
    'Partial Failure': 2,
    'Success': 3
}
df['Status_da_Missao_Num'] = df['Status_da_Missao'].map(mapa_status_missao)

# Mapeamento do Status do Foguete (0 para Aposentado, 1 para Ativo)
mapa_status_foguete = {
    'StatusRetired': 0,
    'StatusActive': 1
}
df['Status_do_Foguete_Num'] = df['Status_do_Foguete'].map(mapa_status_foguete)


In [187]:
df

,Nome_da_Empresa,Status_do_Foguete,Custo_da_Missao,Status_da_Missao,Pais,Estado_Regiao,Ano,Mes,Dia,Hora,Modelo_Foguete,Carga_Util,Pais_Mapa,Status_da_Missao_Num,Status_do_Foguete_Num
0,SpaceX,StatusActive,50.00,Success,USA,Florida,2020,8,7,5,Falcon 9 Block 5,Starlink V1 L9 & BlackSky,United States,3,1
1,CASC,StatusActive,29.75,Success,China,Jiuquan Satellite Launch Center,2020,8,6,4,Long March 2D,Gaofen-9 04 & Q-SAT,China,3,1
2,SpaceX,StatusActive,56.50,Success,USA,Texas,2020,8,4,23,Starship Prototype,150 Meter Hop,United States,3,1
3,Roscosmos,StatusActive,65.00,Success,Kazakhstan,Baikonur Cosmodrome,2020,7,30,21,Proton-M/Briz-M,Ekspress-80 & Ekspress-103,Kazakhstan,3,1
4,ULA,StatusActive,145.00,Success,USA,Florida,2020,7,30,11,Atlas V 541,Perseverance,United States,3,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
4319,US Navy,StatusRetired,150.00,Failure,USA,Florida,1958,2,5,7,Vanguard,Vanguard TV3BU,United States,0,0
4320,AMBA,StatusRetired,150.00,Success,USA,Florida,1958,2,1,3,Juno I,Explorer 1,United States,3,0
4321,US Navy,StatusRetired,150.00,Failure,USA,Florida,1957,12,6,16,Vanguard,Vanguard TV3,United States,0,0
4322,RVSN USSR,StatusRetired,150.00,Success,Kazakhstan,Baikonur Cosmodrome,1957,11,3,2,Sputnik 8K71PS,Sputnik-2,Kazakhstan,3,0


# Dicionário

In [188]:
# Dicionário de Dados

dicionario_colunas = {
    'Nome_da_Empresa':    'Texto (String)  | Nome da agência espacial ou empresa privada responsável pelo lançamento.',
    'Status_do_Foguete':  'Texto (String)  | Estado atual do modelo do foguete (Ativo ou Aposentado).',
    'Status_do_Foguete_Num':'Inteiro (Int64) | Codificação numérica do status do foguete (0: Aposentado | 1: Ativo).',
    'Custo_da_Missao':    'Decimal (Float) | Custo estimado do lançamento em Milhões de Dólares (USD) [Dados Imputados].',
    'Status_da_Missao':   'Texto (String)  | Resultado do lançamento (Success, Failure, Partial Failure, Prelaunch Failure).',
    'Status_da_Missao_Num':'Inteiro (Int64) | Escala numérica de sucesso (0: Falha | 1: Falha Pré-Lançamento | 2: Falha Parcial | 3: Sucesso).',
    'Pais':               'Texto (String)  | País de origem extraído da localização geográfica original.',
    'Pais_Mapa':          'Texto (String)  | Nome do país padronizado em inglês para geolocalização perfeita em Dashboards.',
    'Estado_Regiao':      'Texto (String)  | Estado, Província ou Região específica onde a base de lançamento está localizada.',
    'Ano':                'Inteiro (Int64) | Ano em que a missão foi realizada.',
    'Mes':                'Inteiro (Int64) | Mês do lançamento (valores ausentes preservados em missões muito antigas).',
    'Dia':                'Inteiro (Int64) | Dia do mês do lançamento.',
    'Hora':               'Inteiro (Int64) | Hora aproximada do lançamento no padrão de 24h.',
    'Modelo_Foguete':     'Texto (String)  | Nome/Modelo exato do veículo lançador utilizado.',
    'Carga_Util':         'Texto (String)  | Nome da carga, satélite ou missão que foi enviada ao espaço.'
}

print("="*80)
print("             DICIONÁRIO DE DADOS DEFINITIVO - DATASET ESPACIAL")
print("="*80)
for coluna, descricao in dicionario_colunas.items():
    if coluna in df.columns:
        print(f" Coluna: {coluna.ljust(22)} -> {descricao}")
print("="*80)

             DICIONÁRIO DE DADOS DEFINITIVO - DATASET ESPACIAL
 Coluna: Nome_da_Empresa        -> Texto (String)  | Nome da agência espacial ou empresa privada responsável pelo lançamento.
 Coluna: Status_do_Foguete      -> Texto (String)  | Estado atual do modelo do foguete (Ativo ou Aposentado).
 Coluna: Status_do_Foguete_Num  -> Inteiro (Int64) | Codificação numérica do status do foguete (0: Aposentado | 1: Ativo).
 Coluna: Custo_da_Missao        -> Decimal (Float) | Custo estimado do lançamento em Milhões de Dólares (USD) [Dados Imputados].
 Coluna: Status_da_Missao       -> Texto (String)  | Resultado do lançamento (Success, Failure, Partial Failure, Prelaunch Failure).
 Coluna: Status_da_Missao_Num   -> Inteiro (Int64) | Escala numérica de sucesso (0: Falha | 1: Falha Pré-Lançamento | 2: Falha Parcial | 3: Sucesso).
 Coluna: Pais                   -> Texto (String)  | País de origem extraído da localização geográfica original.
 Coluna: Pais_Mapa              -> Texto (String)  | 

In [189]:
# Salvando a base final com as novas colunas numéricas para o seu Dashboard
df.to_csv('Space_Limpa.csv', index=False, encoding='utf-8')
print("\nBase de dados final atualizada e salva com sucesso!")


Base de dados final atualizada e salva com sucesso!
